src/Rules/Registry.php

```diff
@@ -28,7 +28,9 @@ public function __construct(array $rules)
 	public function getRules(string $nodeType): array
 	{
 		if (!isset($this->cache[$nodeType])) {
-			$parentNodeTypes = [$nodeType] + class_parents($nodeType) + class_implements($nodeType);
+			$classParents = (array) class_parents($nodeType);
+			$classImplementations = (array) class_implements($nodeType);
+			$parentNodeTypes = [$nodeType] + $classParents + $classImplementations;
```

Comment:
IMHO casting to an array both of this methods isn't an issue, because casting `false` to an `array` will result in a `[false]`, and as we check for a "false" rule in line 37, isn't going to enter in the `$rules`.

Reply:
I don't link this, this is asking for trouble and making use of loose casting - it asks for `$this->rules[0]` a few lines below this. Please make the code more explicit - I think the error can occur with an autoloading failure, but the classes here are already autoloaded, so throw `ShouldNotHappenException` in case of false, thanks.

Reply:
See: https://3v4l.org/Or3v1